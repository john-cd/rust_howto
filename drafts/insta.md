# Testing - Insta

[![insta][c-insta-badge]][c-insta]
[![insta-crates.io][c-insta-crates.io-badge]][c-insta-crates.io]
[![insta-github][c-insta-github-badge]][c-insta-github]
[![insta-lib.rs][c-insta-lib.rs-badge]][c-insta-lib.rs]

[![cargo-insta][c-cargo_insta-badge]][c-cargo_insta]
[![cargo-insta-crates.io][c-cargo_insta-crates.io-badge]][c-cargo_insta-crates.io]
[![cargo-insta-github][c-cargo_insta-github-badge]][c-cargo_insta-github]
[![cargo-insta-lib.rs][c-cargo_insta-lib.rs-badge]][c-cargo_insta-lib.rs]

{{#include refs.incl.md}}
{{#include ../../refs/link-refs.md}}

<div class="hidden">
TODO
https://insta.rs/

[c-insta-badge]: https://img.shields.io/crates/v/insta?label=insta
[c-insta-crates.io-badge]: https://img.shields.io/badge/crates.io-insta-crimson
[c-insta-crates.io]: https://crates.io/crates/insta
[c-insta-github-badge]: https://img.shields.io/badge/insta-steelblue?logo=github
[c-insta-lib.rs-badge]: https://img.shields.io/badge/lib.rs-insta-yellow
[c-insta-lib.rs]: https://lib.rs/crates/insta
[c-insta]: https://docs.rs/insta

[c-cargo_insta-badge]: https://img.shields.io/crates/v/cargo-insta?label=cargo-insta
[c-cargo_insta-crates.io-badge]: https://img.shields.io/badge/crates.io-cargo--insta-crimson
[c-cargo_insta-crates.io]: https://crates.io/crates/cargo_insta
[c-cargo_insta-github-badge]: https://img.shields.io/badge/cargo--insta-steelblue?logo=github
[c-cargo_insta-lib.rs-badge]: https://img.shields.io/badge/lib.rs-cargo--insta-yellow
[c-cargo_insta-lib.rs]: https://lib.rs/crates/cargo-insta
[c-cargo_insta]: https://docs.rs/cargo_insta
</div>
