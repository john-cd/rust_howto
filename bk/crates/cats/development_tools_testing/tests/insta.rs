// ANCHOR: example
//! Snapshot testing with the `insta` crate keeps track of how your data
//! structures or output change over time.
use insta::assert_snapshot;

/// This is a sample function that formats user details.
pub fn format_user(name: &str, age: u32) -> String {
    format!("Name: {}, Age: {}", name, age)
}

/// The following is a test that uses `insta` to snapshot the output.
/// Run with `cargo test`.
/// If the test fails, `insta` will prompt you to review and accept the new
/// snapshot. This allows you to easily manage changes to your output
/// over time.
#[test]
fn test_format_user() {
    let formatted = format_user("Alice", 30);
    assert_snapshot!("format_user_snapshot", formatted);
}
// ANCHOR_END: example
