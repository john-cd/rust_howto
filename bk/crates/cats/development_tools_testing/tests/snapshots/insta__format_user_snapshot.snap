---
source: crates/cats/development_tools_testing/tests/insta.rs
expression: formatted
---
Name: Alice, Age: 30
